### PR TITLE
transition file contexts to /run

### PR DESCRIPTION
**Pull request by cgzones**
Remove file context aliases and update file context paths to use the /run filesystem path.
Add backward compatibility file context alias for /var/run using applications like https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=783321

Associated with https://github.com/TresysTechnology/refpolicy-contrib/pull/31

### CODE REVIEW

File: policy/modules/kernel/files.fc [hunk 1]
```diff
@@ -165,6 +165,7 @@ ifdef(`distro_debian',`
 # /run
 #
 /run			-d	gen_context(system_u:object_r:var_run_t,s0-mls_systemhigh)
+/run			-l	gen_context(system_u:object_r:var_run_t,s0)
```
Comment (cgzones) [inline policy/modules/kernel/files.fc]: this is needed, else /var/run will be lableled as default_t

**Comment by doverride** [comment]
On 12/14/2016 11:56 PM, cgzones wrote:
> cgzones commented on this pull request.
> 
> 
> 
>> @@ -165,6 +165,7 @@ ifdef(`distro_debian',`
>  # /run
>  #
>  /run			-d	gen_context(system_u:object_r:var_run_t,s0-mls_systemhigh)
> +/run			-l	gen_context(system_u:object_r:var_run_t,s0)
> 
> this is needed, else /var/run will be lableled as default_t
> 

This patch will likely cause systemd to break with regards to
systemd-tmpfiles and systemd socket units.

Have to you tested it? I have encountered issues with this on both
Fedora as well as Debian.

We should probably see if the people responsible for these configs and
units are willing to accept patches to fix these issues first.

If they are willing to fix them then i suppose we could implement this patch

known issues fedora:


	/usr/lib/tmpfiles.d/pam.conf
	/usr/lib/tmpfiles.d/libselinux.conf

	To solve:

	cp /usr/lib/tmpfiles.d/pam.conf /etc/tmpfiles.d/ && \
		sed -i 's/\/var\/run/\/run/' /etc/tmpfiles.d/pam.conf
	cp /usr/lib/tmpfiles.d/libselinux.conf /etc/tmpfiles.d/ && \
		sed -i 's/\/var\/run/\/run/' /etc/tmpfiles.d/libselinux.conf


known issues debian stretch:



	/usr/lib/tmpfiles.d/sshd.conf

	To solve:

	cp /usr/lib/tmpfiles.d/sshd.conf /etc/tmpfiles.d/ && \
		sed -i 's/\/var\/run/\/run/' /etc/tmpfiles.d/sshd.conf

	/usr/lib/systemd/system/dbus.socket
	/usr/lib/systemd/system/avahi-daemon.socket

	To solve:

	cp /usr/lib/systemd/system/dbus.socket /etc/systemd/system/ && \
		sed -i 's/\/var\/run/\/run/' /etc/systemd/system/dbus.socket
	cp /usr/lib/systemd/system/avahi-daemon.socket /etc/systemd/system/ && \
		sed -i 's/\/var\/run/\/run/' /etc/systemd/system/avahi-daemon.socket

-- 
Key fingerprint = 5F4D 3CDB D3F8 3652 FBD8  02D5 3B6C 5F1D 2C7B 6B02
https://sks-keyservers.net/pks/lookup?op=get&search=0x3B6C5F1D2C7B6B02
Dominick Grift

**Comment by cgzones** [comment]
i know about these two complications on debian:
- /usr/lib/tmpfiles.d/sshd.conf
- /usr/lib/systemd/system/dbus.socket
and both seem to work for me (with the backward compatibility file alias)

**Comment by doverride** [comment]
On 12/15/2016 12:08 AM, cgzones wrote:
> i know about these two complications on debian:
> - /usr/lib/tmpfiles.d/sshd.conf
> - /usr/lib/systemd/system/dbus.socket
> and both seem to work for me (with the backward compatibility file alias)

o i overlooked that compatibility part then. good. nevermind then.

> 


-- 
Key fingerprint = 5F4D 3CDB D3F8 3652 FBD8  02D5 3B6C 5F1D 2C7B 6B02
https://sks-keyservers.net/pks/lookup?op=get&search=0x3B6C5F1D2C7B6B02
Dominick Grift

File: config/file_contexts.subs_dist [hunk 2]
```diff
@@ -12,11 +12,13 @@
 /lib/systemd /usr/lib/systemd
 /lib32 /lib
 /lib64 /lib
-/run /var/run
-/run/lock /var/lock
```
Comment (pebenito) [inline config/file_contexts.subs_dist]: I think we still need to have a backwards compatibility for `/run/lock` to `/var/lock`. Otherwise I think it's ok.

**Comment by cgzones** [comment]
with the new commit, /var/lock is the standard directory and /run/lock is an alias.
FHS is also proposing using /var/lock

**Comment by pebenito** [comment]
Please squash, then I'll merge.

**Comment by cgzones** [comment]
I rebased the branch and squashed the commits.
Btw do file-aliases apply transitively?
From /var/run/lock/foo (over the alias /var/run -> /run) to /run/lock/foo (and then over the alias (run/lock -> /var/lock) to /var/lock/foo?

**Comment by cgzones** [comment]
Any objections?

**Pull request merged**